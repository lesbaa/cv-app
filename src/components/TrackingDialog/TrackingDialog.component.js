import React from 'react'
import PropTypes from 'prop-types'
import AcceptIcon from 'react-feather/dist/icons/thumbs-up'
import DenyIcon from 'react-feather/dist/icons/thumbs-down'
import Dialog from '~/components/Dialog'
import styles from './TrackingDialog.styles'

const TrackingDialog = ({
  dialogVisible,
  acceptTracking,
  denyTracking,
}) => (
  <Dialog
    message="I have a tiny bit of analytics to see how people use this cv. You cool with that?"
    isVisible={dialogVisible}
  >
    <button
      className="button accept"
      onClick={acceptTracking}
    >
      <AcceptIcon />
    </button>
    <button
      className="button"
      onClick={denyTracking}
    >
      <DenyIcon />
    </button>
    <style jsx>{styles}</style>
  </Dialog>
)

TrackingDialog.defaultProps = {

}

TrackingDialog.propTypes = {

}

export default TrackingDialog
