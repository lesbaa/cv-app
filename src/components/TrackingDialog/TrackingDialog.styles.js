import css from 'styled-jsx/css'

export default css`
  .button {
    font-size: 1em;
    display: inline-block;
    border: 1px solid #444;
    border-radius: 0.25em;
    padding: 0.5em 1em;
    text-align: center;
    margin: 0 1em;
    color: #444;
  }
`
