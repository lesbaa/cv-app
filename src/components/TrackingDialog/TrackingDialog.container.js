import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import {
  acceptTracking,
  denyTracking,
} from '~/actionCreators'
import TrackingDialog from './TrackingDialog.component'

const mapStateToProps = ({
  tracking: {
    dialogVisible,
  },
}) => ({
  dialogVisible,
})

const mapDispatchToProps = dispatch => bindActionCreators({
  acceptTracking,
  denyTracking,
}, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TrackingDialog)
