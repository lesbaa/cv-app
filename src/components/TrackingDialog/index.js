import TrackingDialog from './TrackingDialog.container'

export default TrackingDialog
