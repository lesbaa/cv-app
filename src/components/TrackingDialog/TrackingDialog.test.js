/* eslint-env jest */
/* global window */
import React from 'react'
import { shallow } from 'enzyme'
import TrackingDialog from './TrackingDialog.component'
import Dialog from '~/components/Dialog'

jest.mock('~/components/Dialog', () => props => <div className="Dialog" />)

describe('Title', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const mockAcceptTracking = jest.fn()
  const mockDenyTracking = jest.fn()

  const wrapper = shallow(
    <TrackingDialog
      dialogVisible
      acceptTracking={mockAcceptTracking}
      denyTracking={mockDenyTracking}
    />
  )
  const wrapperNoProps = shallow(
    <TrackingDialog />
  )

  it('matches its snapshots', () => {
    expect(wrapper).toMatchSnapshot('with props')
    expect(wrapperNoProps).toMatchSnapshot('with no props')
  })

  it('calls acceptTracking on click of accept', () => {
    const mockEventData = 'ðŸŽ‰'
    wrapper
      .find('.button.accept')
      .simulate('click', mockEventData)
    expect(mockAcceptTracking).toHaveBeenCalledWith(mockEventData)
  })

  it('passes the dialogVisible prop to the isVisible prop of Dialog', () => {
    const dialogIsVisibleProp = wrapper
      .find(Dialog)
      .props()
      .isVisible

    expect(dialogIsVisibleProp).toBe(true)

    const dialogIsNotVisible = wrapperNoProps
      .find(Dialog)
      .props()
      .isVisible

    expect(dialogIsNotVisible).toBe(false)
  })

  it('calls denyTracking on click of deny', () => {
    const mockEventData = 'ðŸ˜¦'
    wrapper
      .find('.button')
      .at(1)
      .simulate('click', mockEventData)
    expect(mockDenyTracking).toHaveBeenCalledWith(mockEventData)
  })

  it('calls console.log if no actions are passed', () => {
    const consoleSpy = jest.spyOn(console, 'log')

    wrapperNoProps
      .find('.button')
      .forEach(node => node.simulate('click'))

    expect(consoleSpy).toHaveBeenCalledWith('TrackingDialog: no acceptTracking action passed as prop')
    expect(consoleSpy).toHaveBeenCalledWith('TrackingDialog: no denyTracking action passed as prop')
  })
})
