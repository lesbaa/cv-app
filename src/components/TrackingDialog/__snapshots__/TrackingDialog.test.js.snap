// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title matches its snapshots: with no props 1`] = `ShallowWrapper {}`;

exports[`Title matches its snapshots: with props 1`] = `ShallowWrapper {}`;
